Observer/UserProfileNewsletterSubscribeObserver.php

```diff
@@ -12,32 +12,36 @@ class UserProfileNewsletterSubscribeObserver implements ObserverInterface
 
     public function __construct(
         \Klaviyo\Reclaim\Helper\Data $data_helper,
-        \Magento\Framework\App\RequestInterface $request,
         \Magento\Customer\Api\CustomerRepositoryInterface $customer_repository_interface
     ) {
-        $this->data_helper = $data_helper;
-        $this->request = $request;
+        $this->data_helper                   = $data_helper;
         $this->customer_repository_interface = $customer_repository_interface;
     }
 
     public function execute(\Magento\Framework\Event\Observer $observer)
     {
-        if (!$this->data_helper->getEnabled()) return;
-
+        if (!$this->data_helper->getEnabled()) {
+            return;
+        }
+        $customer   = null;
         $subscriber = $observer->getDataObject();
 
         if ($subscriber->isStatusChanged()) {
-          $customer = $this->customer_repository_interface->getById($subscriber->getCustomerId());
-
-          if ($subscriber->isSubscribed()) {
-            $this->data_helper->subscribeEmailToKlaviyoList(
-                $customer->getEmail(),
-                $customer->getFirstname(),
-                $customer->getLastname()
-            );
-          } else {
-            $this->data_helper->unsubscribeEmailFromKlaviyoList($customer->getEmail());
-          }
+            if ($subscriber->getCustomerId()) {
+                $customer = $this->customer_repository_interface->getById($subscriber->getCustomerId());
```

Comment:
looks like we don't do anything if customer is null could we return early?

Reply:
If customer is null we get the email from a subscriber